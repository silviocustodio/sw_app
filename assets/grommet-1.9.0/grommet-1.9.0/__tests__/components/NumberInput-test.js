// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import NumberInput from '../../src/js/components/NumberInput';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('NumberInput', () => {
  it('has correct default options', () => {
    const component = renderer.create(
       <NumberInput id="item2" name="item2" value={10} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
