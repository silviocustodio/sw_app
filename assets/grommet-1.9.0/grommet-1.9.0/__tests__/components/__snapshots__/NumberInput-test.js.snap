// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput has correct default options 1`] = `
<span
  className="grommetux-number-input"
>
  <input
    className="grommetux-input grommetux-number-input__input"
    disabled={undefined}
    id="item2"
    name="item2"
    tabIndex="0"
    type="number"
    value={10}
  />
  <button
    aria-label={undefined}
    className="grommetux-button grommetux-button--plain grommetux-number-input__subtract"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="grommetux-button__icon"
    >
      <svg
        aria-label="subtract"
        className="grommetux-control-icon grommetux-control-icon-subtract grommetux-control-icon--responsive"
        height="24px"
        role="img"
        version="1.1"
        viewBox="0 0 24 24"
        width="24px"
      >
        <path
          d="M2,12 L22,12"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </span>
  </button>
  <button
    aria-label={undefined}
    className="grommetux-button grommetux-button--plain grommetux-number-input__add"
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    type="button"
  >
    <span
      className="grommetux-button__icon"
    >
      <svg
        aria-label="add"
        className="grommetux-control-icon grommetux-control-icon-add grommetux-control-icon--responsive"
        height="24px"
        role="img"
        version="1.1"
        viewBox="0 0 24 24"
        width="24px"
      >
        <path
          d="M12,22 L12,2 M2,12 L22,12"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </span>
  </button>
</span>
`;
